File: c#/EmployeeManagement/Models/FTInfoTech.cs
namespace EmployeeManagement.Models
{
    public class FTInfoTech : FTEmployee
    {
        public string OnCallGroup { get; }

        public FTInfoTech(int id, string firstName, string lastName, DateOnly dateOfBirth, string position, Departments department, double salary, string onCallGroup)
            : base(id, firstName, lastName, dateOfBirth, position, department, salary)
        {
            OnCallGroup = !string.IsNullOrWhiteSpace(onCallGroup)
                ? onCallGroup
                : throw new System.ArgumentException("On-call group cannot be null or empty.", nameof(onCallGroup));
        }

        public override void ReportToManager()
        {
            Console.WriteLine($"FT InfoTech {FirstName} {LastName} reports to the manager and is assigned to group {OnCallGroup}.");
        }

        public override string ToString()
        {
            return base.ToString() + $", OnCallGroup={OnCallGroup}";
        }
    }
}

File: Java/FTInfoTech.java
public class FTInfoTech extends FTEmployee {

    private String on_call_group;

    public FTInfoTech(
        int id,
        String first_name,
        String last_name,
        String dob,
        String position,
        Departments department,
        double salary,
        String on_call_group
    ) {
        super(id, first_name, last_name, dob, position, department, salary);
        this.on_call_group = on_call_group;
    }

    @Override
    public void reportToManager() {
    }
}
